workerPool.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+)
+
+//worker blocks until the object ids are available, and sends gotten details to result
+func (req *client) worker(ctx context.Context, jobs <-chan []int, seen map[int]struct{}, result chan<- ObjectDetail) {
```

Comment:
You have different recievers. `c` in one method and `req` in another one. They should be the same